core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestion.java

```diff
@@ -253,6 +263,58 @@ protected XContentBuilder innerToXContent(XContentBuilder builder, Params params
                 return builder;
             }
 
+            private static ObjectParser<Map<String, Object>, Void> PARSER = new ObjectParser<>("CompletionOptionParser",
+                    true, HashMap::new);
+
+            static {
+                SearchHit.declareInnerHitsParseFields(PARSER);
+                PARSER.declareString((map, value) -> map.put(Suggestion.Entry.Option.TEXT.getPreferredName(), value),
+                        Suggestion.Entry.Option.TEXT);
+                PARSER.declareFloat((map, value) -> map.put(Suggestion.Entry.Option.SCORE.getPreferredName(), value),
+                        Suggestion.Entry.Option.SCORE);
+                PARSER.declareObject((map, value) -> map.put(CompletionSuggestion.Entry.Option.CONTEXT.getPreferredName(), value),
+                        Option::parseContext, CompletionSuggestion.Entry.Option.CONTEXT);
+            }
+
+            private static Map<String, Set<CharSequence>> parseContext(XContentParser parser, Void context) throws IOException {
+                Map<String, Set<CharSequence>> contexts = new HashMap<>();
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+                while((parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
+                    String key = parser.currentName();
+                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser::getTokenLocation);
+                    Set<CharSequence> values = new HashSet<>();
+                    for (Object value : parser.list()) {
```

Comment:
I think we should parse the values using a `while((parser.nextToken()) != XContentParser.Token.END_ARRAY) ` loop and check the token type for each value. We expect only strings and we should throw an exception if we found something else